pkg/extensions/search/utils/db.go

```diff
@@ -0,0 +1,229 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/boltdb/bolt"
+)
+
+const (
+	// NvdDB ...
+	NvdDB = "NvdJSON"
+	// VendorDB ...
+	VendorDB = "NvdPkgVendor"
+	// NameDB ...
+	NameDB = "NvdPkgName"
+	// NameverDB ...
+	NameverDB = "NvdPkgNameVer"
+	// NvdmetaDB ...
+	NvdmetaDB = "NvdMeta"
+)
+
+/*Conn ... Create a database connection to */
+func Conn(dbPath string) *bolt.DB {
+	// Opening the connection on DB on given port
+	db, err := bolt.Open(dbPath, 0600, nil)
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+
+	return db
+}
+
+//CreateDB ...
+func CreateDB(dbname string, db *bolt.DB) bool {
+	// Creating the bucket on already open DB
+	uerr := db.Update(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists([]byte(dbname))
+		if err != nil {
+			return fmt.Errorf("create bucket: %s", err)
+		}
+		return nil
+	})
+
+	return uerr == nil
+}
+
+func updateNVD(schemas []Schema, mapList []map[string][]CVEId, db *bolt.DB) bool {
+	err := db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(NvdDB))
+
+		for _, schema := range schemas {
+			encoded, err := json.Marshal(schema)
+			if err != nil {
+				return err
+			}
+
+			err = b.Put([]byte(schema.CveID), encoded)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+
+	uerr := updateNVDPkg(VendorDB, mapList[0], db)
+	if uerr != nil {
+		fmt.Println("Unable to Update Vendor Package Bucket")
+		return false
+	}
+
+	uerr = updateNVDPkg(NameDB, mapList[1], db)
+	if uerr != nil {
+		fmt.Println("Unable to Update Name Package Bucket")
+
+		return false
+	}
+
+	uerr = updateNVDPkg(NameverDB, mapList[2], db)
+	if uerr != nil {
+		fmt.Println("Unable to Update Name-Version Package Bucket")
```

Comment:
these need to be logs

Reply:
I will change all the information messages/errors  in form of logs.